sql/parser/normalize.go

```diff
@@ -99,6 +99,16 @@ func (v *normalizeVisitor) Visit(expr Expr, pre bool) (Visitor, Expr) {
 
 	// Evaluate constant expressions.
 	if isConst(expr) {
+		// Normalize constant In and NotIn comparison expressions with DTuple datum.
+		if cmp, ok := expr.(*ComparisonExpr); ok {
+			switch cmp.Operator {
+			case In, NotIn:
+				tuple := cmp.Right.(DTuple)
```

Comment:
`DTuple` is a `[]Datum`, so this is making a copy of the slice header. That doesn't sound problematic. I doubt `cmd.Right.(DTuple).Normalize()` would work because `Normalize` operates on a `*DTuple`.